package NYT.NChunkServer.NProto;

import "yt_proto/yt/core/misc/proto/guid.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqStartMergeTransaction
{ }

message TReqCreateChunks
{
    required NYT.NProto.TGuid transaction_id = 1;

    message TSubrequest
    {
        required string account = 1;
        required int32 type = 2; // NObjectClient::EObjectType
        required int32 replication_factor = 3;
        required int32 erasure_codec = 4; // NErasure::ECodec
        optional bool movable = 5 [default = true];
        required bool vital = 6;
        required int32 medium_index = 7;
        required NYT.NProto.TGuid job_id = 9;

        reserved 8;
    }

    repeated TSubrequest subrequests = 2;
}

message TReqReplaceChunks
{
    message TReplaceChunkInfo
    {
        repeated NYT.NProto.TGuid old_chunk_ids = 1;
        required NYT.NProto.TGuid new_chunk_id = 2;
    }

    repeated TReplaceChunkInfo replacements = 1;

    required NYT.NProto.TGuid node_id = 2;
    required NYT.NProto.TGuid chunk_list_id = 3;
}

message TReqFinalizeChunkMergeSessions
{
    message TSessionResult
    {
        required NYT.NProto.TGuid node_id = 1;
        required int32 result = 2; // EMergeSessionResult
    }

    repeated TSessionResult subrequests = 1;
}

////////////////////////////////////////////////////////////////////////////////
